presto-spi/src/main/java/com/facebook/presto/spi/block/BlockBuilder.java

```diff
@@ -94,4 +94,12 @@ default BlockBuilder beginBlockEntry()
      * Resets the block builder, clearing all of the data.
      */
     void reset(BlockBuilderStatus blockBuilderStatus);
+
+    /**
+     * This method is similar to the reset method except that
```

Comment:
`Creates a new block builder of the same type based on the current usage statistics of this block builder.`